rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ReportsJdbcRepository.scala

```diff
@@ -449,18 +462,42 @@ class ReportsJdbcRepository(jdbcTemplate : JdbcTemplate) extends ReportsReposito
     }
   }
 
+
   override def getChangeReportsOnInterval(lowestId: Long, highestId: Long): Box[Seq[ResultRepairedReport]] = {
-    val query = s"${baseQuery} and eventtype='${Reports.RESULT_REPAIRED}' and id >= ${lowestId} and id <= ${highestId} order by executionTimeStamp asc"
-    try {
-      Full(jdbcTemplate.query(query,ReportsMapper).asScala.collect{case r:ResultRepairedReport => r})
-    } catch {
-      case ex: Exception =>
-        val error = Failure("Error when trying to retrieve change reports", Some(ex), Empty)
-        logger.error(error)
-        error
+    def theQueryOnly = {
+      val query = s"${baseQuery} and eventtype='${Reports.RESULT_REPAIRED}' and id >= ${lowestId} and id <= ${highestId} order by executionTimeStamp asc"
+      jdbcTemplate.query(query,ReportsMapper)
     }
+
+    transformJavaList(theQueryOnly)
   }
 
+  override def getChangeReportsByRuleOnInterval(ruleId: RuleId, interval: Interval, limit: Option[Int]): Box[Seq[ResultRepairedReport]] = {
+    def theQueryOnly = {
+      val l = limit match {
+        case Some(i) if(i > 0) => s"limit ${i}"
+        case _                 => ""
+      }
+      val query = s"${baseQuery} and eventtype='${Reports.RESULT_REPAIRED}' and ruleid='${ruleId.value}' " +
+                  s" and executionTimeStamp >  '${new Timestamp(interval.getStartMillis)}'::timestamp " +
+                  s" and executionTimeStamp <= '${new Timestamp(interval.getEndMillis)  }'::timestamp order by executionTimeStamp asc ${l}"
+      jdbcTemplate.query(query,ReportsMapper)
+    }
+
+    transformJavaList(theQueryOnly)
+  }
+
+  private[this] def transformJavaList(l: java.util.List[Reports]) = {
```

Comment:
that was mesured, and even if it is the case (not exactly the case, not sur why), it's negligible compared to other hot spot